pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java

```diff
@@ -2813,21 +2824,30 @@ public void removedConsumer(Consumer consumer) {
 
     @Override
     public void removedProducer(Producer producer) {
-        safelyRemoveProducer(producer);
+        safelyRemoveProducer(producer, false);
     }
 
-    private void safelyRemoveProducer(Producer producer) {
+    private void safelyRemoveProducer(Producer producer, boolean keepTombstoneForProducer) {
         long producerId = producer.getProducerId();
         if (log.isDebugEnabled()) {
             log.debug("[{}] Removed producer: producerId={}, producer={}", remoteAddress, producerId, producer);
         }
```

Comment:
Not really a problem in this PR, but this log line isn't accurate since it happens before anything got removed.

Reply:
I agree, but I'll leave that for another PR. I think there is an easy refactor of the producers map to make it so we only use it from the event loop, and that will clean up this code.